includes/utilities.inc

```diff
@@ -70,29 +75,26 @@ function islandora_gsearcher_speak_to_gsearch($pid, $opp) {
     $verb = 'purge';
   }
 
-  $password = variable_get('islandora_gsearcher_gsearch_pass', 'fedoraAdmin');
-  $user = variable_get('islandora_gsearcher_gsearch_user', 'fedoraAdmin');
-  $url = variable_get('islandora_gsearcher_gsearch_url', 'localhost:8080/fedoragsearch/rest');
+  $password = $config->get('islandora_gsearcher_gsearch_pass');
+  $user = $config->get('islandora_gsearcher_gsearch_user');
+  $url = $config->get('islandora_gsearcher_gsearch_url');
 
-  $response = drupal_http_request(
-    url(
-      "http://$user:$password@$url",
-      array(
-        'query' => array(
-          'operation' => 'updateIndex',
-          'action' => $action,
-          'value' => $pid,
-        ),
-      )
-    )
-  );
-  // Ping the logs in case of failure, GSearch will know more.
-  if ($response->code != 200) {
-    watchdog(
-      'Islandora GSearcher',
-      'GSearch did not return 200 when asked to @verb @pid.',
-      array('@verb' => $verb, '@pid' => $pid),
-      WATCHDOG_ERROR
-    );
+  $client = \Drupal::httpClient();
+  $url_params = [
+    'query' => [
+      'operation' => 'updateIndex',
+      'action' => $action,
+      'value' => $pid,
+    ],
+  ];
+  $url = Url::fromUri("http://$user:$password@$url", $url_params);
+  try {
+    $client->request('GET', $url->toString());
+  }
+  catch (RequestException $exception) {
```

Comment:
The error handling and reporting needs to be updated here in a similar fashion to the admin page.